src/server.ts

```diff
@@ -1,47 +1,84 @@
 import * as fsPath from 'path';
+import * as http from 'http';
 import * as nunjucks from 'nunjucks';
 
 import {Pod} from './pod';
 import {StaticRoute} from './router';
 
 import express = require('express');
 
-export class Server {
-  constructor() {}
+interface ServerOptions {
+  port: string | number;
 }
 
-export function createApp(pod: Pod) {
-  const app = express();
-  app.disable('x-powered-by');
-  pod.plugins.trigger('createServer', app);
-  app.all('/*', async (req: express.Request, res: express.Response) => {
-    try {
-      const route = pod.router.resolve(req.path);
-      if (!route) {
-        res
-          .status(404)
-          .sendFile(fsPath.join(__dirname, './static/', 'error-no-route.html'));
-        return;
-      }
-      if (route.provider.type === 'static_dir') {
-        res.sendFile(
-          pod.getAbsoluteFilePath((route as StaticRoute).staticFile.podPath)
-        );
-        return;
-      } else {
-        const content = await route.build();
-        res.set('Content-Type', route.contentType);
-        res.send(content);
+export class Server {
+  private pod: Pod;
+  httpServer?: http.Server;
```

Comment:
per comment above, i think this should be private since it's an implementation detail that a caller should not have access to.

Reply:
It's private now